instagram-app/src/App.js

```diff
@@ -0,0 +1,47 @@
+import React, { Component } from 'react';
+import './App.css';
+import dummyData from './dummy-data';
+import PostsContainer from './components/PostContainer/PostsContainer';
+import SearchBar from './components/SearchBar/SearchBarContainer';
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      posts: [],
+      filteredPosts: []
+    };
+  }
+
+componentDidMount() {
+  this.setState({ posts: dummyData });
+}
+
+searchPostsHandler = e => {
+  const posts = this.state.posts.filter(p => {
+    if (p.p.username.includes(e.target.value)) {
```

Comment:
I think you have an extra p on this? Your search functionality is not currently returning filtered posts. But this is on the right track. Another option is to control the input with your state. ie. put a "searchInput" variable on the state and have a handleSearchInput method.